components/MyLeafletMap.vue

```diff
@@ -67,8 +66,11 @@ function generateLabels(data: GeoJSON.FeatureCollection): Map<string, string> {
   );
 
   if (legendDisplayOption[0] === 'default') {
-    uniqueValues.forEach((value, i) => {
-      colorMap.set(value, generateColor(i, uniqueValues.length));
```

Comment:
@alifarooq93 Are we still displaying "default" for those markers that do not have a value matching the configured legend details? That would be important. And maybe we should change default to something like `unexpected value`